hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -243,17 +318,34 @@ public TokenRelationship newRelationshipWith(final Account account) {
 		return newRel;
 	}
 
-	private void changeSupply(TokenRelationship treasuryRel, long amount, ResponseCodeEnum negSupplyCode) {
+	/**
+	 * Creates new {@link TokenRelationship} for the specified {@link Account}
+	 * IMPORTANT: The provided account is set to KYC granted and unfrozen by default
+	 * @param account the Account for which to create the relationship
+	 * @return newly created {@link TokenRelationship}
+	 */
+	public TokenRelationship newEnabledRelationship(final Account account) {
+		final var rel = new TokenRelationship(this, account);
+		if (hasKycKey()) {
+			rel.changeKycState(true);
+		}
+		if (hasFreezeKey()) {
+			rel.changeFrozenState(false);
+		}
+		return rel;
+	}
+
+	private void changeSupply(TokenRelationship treasuryRel, long amount, ResponseCodeEnum negSupplyCode, boolean ignoreSupplyKey) {
 		validateTrue(treasuryRel != null, FAIL_INVALID,
 				"Cannot mint with a null treasuryRel");
 		validateTrue(treasuryRel.hasInvolvedIds(id, treasury.getId()), FAIL_INVALID,
 				"Cannot change " + this + " supply (" + amount + ") with non-treasury rel " + treasuryRel);
-
-		validateTrue(supplyKey != null, TOKEN_HAS_NO_SUPPLY_KEY);
-
+		if (!ignoreSupplyKey) {
+			validateTrue(supplyKey != null, TOKEN_HAS_NO_SUPPLY_KEY);
+		}
 		final long newTotalSupply = totalSupply + amount;
 		validateTrue(newTotalSupply >= 0, negSupplyCode);
-
+		// TODO check what is the response if the initial supply is more than the max supply upon creation
```

Comment:
Remove the TODO if you already checked it

Reply:
`INVALID_TOKEN_INITIAL_SUPPLY` - i think this is the correct one

Reply:
Does this mean that we should change the implementation of `changeSupply?`?